hack/verify.sh

```diff
@@ -24,12 +24,22 @@ cd "${REPO_ROOT}"
 header_text "running generate"
 make generate
 
-header_text "running golangci-lint"
-make lint
+# Only run verify-generate in CI, otherwise running generate
+# locally (which is a valid operation) causes `make test` to fail.
+if [[ -n ${CI} ]]; then
+    header_text "verifying generate"
+    make verify-generate
+fi
 
-# Only run module verification in CI, otherwise updating
+header_text "running modules"
```

Comment:
Flipped the order here so we first run go mod tidy before running the linter.

Depending on the state of the go.mod files the linter could fail because they are out-of-date.